lms/static/scripts/frontend_apps/components/dashboard/DashboardActivityFilters.tsx

```diff
@@ -12,6 +16,7 @@ export type DashboardActivityFiltersProps = {
   onAssignmentsChange: (newAssignmentIds: string[]) => void;
   selectedStudentIds: string[];
   onStudentsChange: (newStudentIds: string[]) => void;
+  resetSelection?: () => void;
```

Comment:
At first I was planning to simply do something in these lines:

```ts
onClick={() => {
  onCoursesChange([]);
  onAssignmentsChange([]);
  onStudentsChange([]);
}}
```

However, since all these end up calling `useDashboardFilters`' updateFilters, which does not use regular state that can be "queued" by preact, only the last one ended up being applied.

Instead, I added an optional extra callback, so that parent components can decide what to do to clear filters, which in the case of persisting in the query string is `updateFilters({ studentIds: [], assignmentIds: [], courseIds: [] })`.